appveyor s3 support

rhyolight opened the pull request:
Fixes #311 (not yet, but eventually!)

I'm taking this over from @rcrowder (thanks!) and I'll be working on this tomorrow. Builds will probably break. I'll send a message about it when I start futzing around. 

This PR is a work in progress!

rcrowder commented:
Not sure what speed difference will be when they move from a test image. http://www.appveyor.com/blog/2015/01/20/visual-studio-2015-ctp-image

rcrowder commented:
:tada: Party on dude! :clap:

rhyolight commented:
Doing remaining tweaks on master branch. Looks like AppVeyor says everything went well, but I don't see the zip file uploaded to S3. I just added an AWS region to the config and changed the upload location, we'll see what happens in https://ci.appveyor.com/project/numenta-ci/nupic-core/build/0.3.0.148
